### DISCUSSION TITLE
Vercel cache miss when using signed s3 URLs

### DISCUSSION THREAD

**Discussion by michaelaflores** [vercel/community#886] [Help] [answered]
Vercel images are always loaded with a cache miss header when using signed S3 URLs. I'm wondering if there's a way to set the Vercel cache key to the image resource identifier rather than using the entire URL including query params?

**Comment by amyegan** [comment, marked as answer]
Hi @michaelaflores! Can you share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) (or your repo if it is public) so I can get a better understanding of the problem you're facing?

**Reply by michaelaflores** [reply]
Hey @amyegan :wave: I think this may be a fair case where a minimal repro isn't quite feasible since I'd need to also set up a mechanism to generate signed S3 URLs solely for this example (requiring at a minimum a few AWS resources, and an endpoint to do the signing). I'm hoping the below may help illustrate the problem:

1. When a request is made for an image that doesn't require authentication to access, the response header is either `x-vercel-cache: STALE` or `x-vercel-cache: HIT`. Example URL format `https://my.app/_next/image?url=https%3A%2F%2Fcontent.my.app%2Fassets%2Favatars%2Favatar-generic-thumbnail.png&w=828&q=75`
2. When a request is made for an image that isn't publicly accessible, the response header is always `x-vercel-cache: MISS`. Example URL format `https://my.app/_next/image?url=https%3A%2F%2Fmy-app.s3.us-east-2.amazonaws.com%2F4ca-04c9%2Fposter.jpg%3FX-Amz-Algorithm%HMAZ-SHA256%26X-Amz-Credential%2022%252Fus-west-2%252Fs3%252Faws4_request%26X-Amz-Date%3D20220922T150912Z%26X-Amz-Expires%3D3600%26X-Amz-Security-Token%TOKEN%252F%%26X-Amz-SignedHeaders%3Dhost%0X-Amz-Signature%SIG&w=828&q=75` The `url` value here is always teh same in this example image, but of course the `Amz` headers change from request to request. Since the _resource_ is the same I'm wondering if there's a way to essentially parse out the actual image URL and have Vercel recognize it as the cache key for the request.

**Reply by amyegan** [reply]
Totally fair to not have a fully working example to share with AWS resources, but thank you for sharing this detailed walk-through of your problem.

My gut says that we wouldn't want the second example to be cached for security reasons. After all, there must be a reason for requiring the security token in the first place. 😄 But I'm going to ask the team and see if there is a way to ignore the headers and cache the resource. I'll update here when I know more.

**Reply by amyegan** [reply]
There are some possibilities with a custom solution that would allow you to cache these resources. This Stack Overflow post demonstrates a few ideas: https://stackoverflow.com/questions/19583338/s3-signed-urls-and-caching

**Reply by michaelaflores** [reply]
@amyegan get out of my brain, I literally was reading that post just now haha. On some reflection something along these lines does seem more correct.
